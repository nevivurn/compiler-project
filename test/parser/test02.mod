//
// test02
//
// parser test:
// - global variables
// - module body
// - assignments
//

module test02;

var a   : integer;
    b,c : integer;

begin
  a := 1;
  b := 2;
  c := 3
end test02.
