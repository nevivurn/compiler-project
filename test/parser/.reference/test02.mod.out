parsing 'test02.mod'...
successfully parsed.
  AST:
    CAstScope: 'test02'
      symbol table:
        [[
          [ *DIM(<ptr(8) to <NULL>>,<integer>) --> <integer>           ]
          [ *DOFS(<ptr(8) to <NULL>>) --> <integer>           ]
          [ *ReadInt() --> <integer>           ]
          [ *ReadLong() --> <longint>           ]
          [ *WriteChar(<char>) --> <NULL>           ]
          [ *WriteInt(<integer>) --> <NULL>           ]
          [ *WriteLn() --> <NULL>           ]
          [ *WriteLong(<longint>) --> <NULL>           ]
          [ *WriteStr(<ptr(8) to array of <char>>>) --> <NULL>           ]
          [ @a        <integer>           a           ]
          [ @b        <integer>           b           ]
          [ @c        <integer>           c           ]
          [ main     <NULL>           ]
        ]]
      statement list:
        := <integer>
          [ @a        <integer>           a ] <integer>
          1 <integer>
        := <integer>
          [ @b        <integer>           b ] <integer>
          2 <integer>
        := <integer>
          [ @c        <integer>           c ] <integer>
          3 <integer>
      nested scopes:
        empty.
    


run the following command to convert the .dot file into a PDF:
  dot -Tpdf -otest02.mod.ast.pdf test02.mod.ast.dot


Done.
